src/gsi/guess_grids.F90

```diff
@@ -977,18 +977,30 @@ subroutine create_gesfinfo
     nfldaer_all=nfldaer
     nfldaer_now=0
     extrap_intime=.true.
-    allocate(hrdifsfc(nfldsfc),ifilesfc(nfldsfc), &
-             hrdifnst(nfldnst),ifilenst(nfldnst), &
-             hrdifsig(nfldsig),ifilesig(nfldsig), &
-             hrdifaer(nfldaer),ifileaer(nfldaer), &
-	     hrdifsfc_all(nfldsfc_all), &
-             hrdifnst_all(nfldnst_all), &
-	     hrdifsig_all(nfldsig_all), &
-             hrdifaer_all(nfldaer_all), &
-	     stat=istatus)
+    if(nfldsig>0) allocate(hrdifsig(nfldsig),ifilesig(nfldsig), &
+                           hrdifsig_all(nfldsig_all), &
+                           stat=istatus)
+    if (istatus/=0) &
```

Comment:
General comment which applies here and elsewhere in `guess_grids.F90`.  

If allocating these arrays generates an error, is it sufficient to simply print a message and continue program execution or should program execution terminate?   

I ask this question given NCO's request for improved error handling in operational codes.

Reply:
This one slipped through the cracks.  I have addressed it by calling `die`.  The PR is open to Ming's branch presently (hu5970/GSI#10).

Reply:
Actually, this was fixed already with hu5970#9.